isort/core.py

```diff
@@ -68,6 +70,10 @@ def process(
     stripped_line: str = ""
     end_of_file: bool = False
     verbose_output: List[str] = []
+    all_imports: List[str] = []
+    if imports_only:
+        _output_stream = output_stream
+        output_stream = open(os.devnull, "wt")
```

Comment:
DeepSource doesn't like this line because ```open``` without ```with``` block. Can change it to ```output_stream = StringIO()``` but it will be less efficient in big files: all stream would accumulate in memory.

Reply:
another possible option
```python
        class DevNull:
            def write(self, *a, **kw):
                pass

        output_stream = DevNull()
```

Reply:
I like that option, I'd even go so far as:

```python
class DevNull:
    def write(*a, **kw):
        pass

output_stream = DevNull
```

Reply:
This works, but mypy isn't happy about type of ```output_stream``` =)
Pushed fix, that satisfies linters and should work fine.